src/script/cpp_api/s_base.cpp

```diff
@@ -139,7 +146,10 @@ bool ScriptApiBase::loadScript(const std::string &scriptpath)
 		errorstream << "Failed to load and run script from " << std::endl;
 		errorstream << scriptpath << ":" << std::endl;
 		errorstream << std::endl;
-		errorstream << lua_tostring(L, -1) << std::endl;
+		std::string errormsg = lua_tostring(L, -1);
+		if (error)
+			(*error) = errormsg;
```

Comment:
Sticking this in the middle of a group of logging statements is ugly.  While you're at it you can make it only refference the stream once (as in:
```C++
foostream << "abc" <<std::endl
	<< "def" << std::endl;
```
).